x/evm/types/msg_vote_confirm_gateway_tx.go

```diff
@@ -0,0 +1,124 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+
+	"github.com/axelarnetwork/axelar-core/utils"
+	vote "github.com/axelarnetwork/axelar-core/x/vote/exported"
+)
+
+// NewVoteConfirmGatewayTxRequest creates a message of type VoteConfirmGatewayTxRequest
+func NewVoteConfirmGatewayTxRequest(sender sdk.AccAddress, pollKey vote.PollKey, vote VoteConfirmGatewayTxRequest_Vote) *VoteConfirmGatewayTxRequest {
+	return &VoteConfirmGatewayTxRequest{
+		Sender:  sender,
+		PollKey: pollKey,
+		Vote:    vote,
+	}
+}
+
+// Route implements sdk.Msg
+func (m VoteConfirmGatewayTxRequest) Route() string {
+	return RouterKey
+}
+
+// Type implements sdk.Msg
+func (m VoteConfirmGatewayTxRequest) Type() string {
+	return "VoteConfirmGatewayTx"
+}
+
+// GetChain returns chain in poll key
+func (m VoteConfirmGatewayTxRequest) GetChain() string {
+	pollKeyIdItems := strings.Split(m.PollKey.ID, "_")
+	if len(pollKeyIdItems) != 2 {
+		return ""
+	}
+
+	return pollKeyIdItems[0]
+}
+
+// GetTxID returns the tx id in poll key
+func (m VoteConfirmGatewayTxRequest) GetTxID() string {
+	pollKeyIdItems := strings.Split(m.PollKey.ID, "_")
+	if len(pollKeyIdItems) != 2 {
+		return ""
+	}
```

Comment:
why not just store chain and txID too? then we'll have less of a hard time removing meaning from the poll iD

Reply:
Let's deal this later